### PR TITLE
wall destruction

### PR DESCRIPTION
**Pull request by Emmett-Byrne**

### CODE REVIEW

File: ARGO/include/Debug.h [hunk 1]
```diff
@@ -3,8 +3,8 @@
 #ifdef _DEBUG
 
 #define HP_SYS_DEBUG
-#define INPUT_SYS_DEBUG
+//#define INPUT_SYS_DEBUG
```
Comment (CathalRedmond) [inline ARGO/include/Debug.h:6]: any reason this is commented out?

File: ARGO/src/AudioManager.cpp [hunk 2]
```diff
@@ -104,7 +104,7 @@ void AudioManager::PlaySfxAtPosition(std::string t_filename, glm::vec2 t_sfxPosi
 	}
 
 #ifdef AUDIO_SYS_DEBUG
-	printf("Sound produced at position: [%s, %s] at angle: %s\n", std::to_string(relVec.x).c_str(), std::to_string(relVec.y).c_str(), std::to_string(angle).c_str());
+	//printf("Sound produced at position: [%s, %s] at angle: %s\n", std::to_string(relVec.x).c_str(), std::to_string(relVec.y).c_str(), std::to_string(angle).c_str());
```
Comment (CathalRedmond) [inline ARGO/src/AudioManager.cpp]: any reason this commented out?

File: ARGO/src/AudioManager.cpp [hunk 3]
```diff
@@ -197,7 +197,8 @@ void AudioManager::PlayPlayerFireSfx(std::string t_filename, glm::vec2 t_playerP
 	}
 
 #ifdef AUDIO_SYS_DEBUG
-	std::printf("Playing Player Fire Sound %s on channel %s\n", t_filename.c_str(), std::to_string(channel).c_str());
+	//std::printf("Playing Player Fire Sound %s on channel %s\n", t_filename.c_str(), std::to_string(channel).c_str());
```
Comment (CathalRedmond) [inline ARGO/src/AudioManager.cpp:200]: can you remove this comment

File: ARGO/src/CollisionSystem.cpp [hunk 4]
```diff
@@ -82,6 +83,65 @@ bool CollisionSystem::circleToAABBCollision(Entity* t_entityCircle, Entity* t_en
 	return (deltaX * deltaX + deltaY * deltaY) < (radius * radius);
 }
 
+bool CollisionSystem::edgeOfTheWorldCircle(Entity& t_entity)
+{
+	TransformComponent* position = static_cast<TransformComponent*>(t_entity.getComponent(ComponentType::Transform));
+	int radius = static_cast<ColliderCircleComponent*>(t_entity.getComponent(ComponentType::ColliderCircle))->getRadius();
+	if (position->getPos().x - radius < 0)
+	{
+		Tag tag = static_cast<TagComponent*>(t_entity.getComponent(ComponentType::Tag))->getTag();
```
Comment (CathalRedmond) [inline ARGO/src/CollisionSystem.cpp]: You have code repeated in each of the if statements. If you can cleanup a bit it would be nice. If not thats fine

**Review by CathalRedmond — CHANGES_REQUESTED** [review]

File: ARGO/include/CollisionSystem.h [hunk 5]
```diff
@@ -18,6 +18,7 @@ class CollisionSystem : public BaseSystem
 private:
 	bool circleToCircleCollision(Entity* t_entityCircle1, Entity* t_entityCircle2);
 	bool circleToAABBCollision(Entity* t_entityCircle, Entity* t_entityAABB);
+	bool edgeOfTheWorldCircle(Entity& t_entity);
```
Comment (PrzemekTomczyk) [inline ARGO/include/CollisionSystem.h]: Could you change this to be a bit less ambiguous?

File: ARGO/src/AudioManager.cpp (hunk 3 above)
Comment (PrzemekTomczyk) [reply ARGO/src/AudioManager.cpp:200]: this should stay for debugging

Comment (PrzemekTomczyk) [reply ARGO/src/AudioManager.cpp:200]: same, needs to stay for debugging

File: ARGO/src/CollisionSystem.cpp (hunk 4 above)
Comment (PrzemekTomczyk) [reply ARGO/src/CollisionSystem.cpp]: I agree, this could be cleaned up or put in a function maybe

File: ARGO/src/Game.cpp [hunk 6]
```diff
@@ -374,7 +403,7 @@ void Game::removeDeadEnemies()
 
 void Game::playerFireSound(const createBulletEvent& t_event)
 {
-	m_audioMgr->PlayPlayerFireSfx(Utilities::GUN_FIRE_PATH + "launcher.wav", static_cast<TransformComponent*>(t_event.entity.getComponent(ComponentType::Transform))->getPos(), m_renderSystem.getFocus());
+	//m_audioMgr->PlayPlayerFireSfx(Utilities::GUN_FIRE_PATH + "launcher.wav", static_cast<TransformComponent*>(t_event.entity.getComponent(ComponentType::Transform))->getPos(), m_renderSystem.getFocus());
```
Comment (PrzemekTomczyk) [inline ARGO/src/Game.cpp:373]: This is getting removed in a different branch so its ok to be commented out

**Review by PrzemekTomczyk — CHANGES_REQUESTED** [review]

File: ARGO/src/Game.cpp [hunk 7]
```diff
@@ -114,19 +102,27 @@ Game::~Game()
 /// </summary>
 void Game::run()
 {
-	m_lastTick = SDL_GetTicks();
-	m_lastRender = SDL_GetTicks();
+	Uint32 timePerFrame = 1000 / 60;
```
Comment (CathalRedmond) [inline ARGO/src/Game.cpp:105]: Magic number

File: ARGO/src/PhysicsSystem.cpp [hunk 8]
```diff
@@ -11,7 +11,7 @@ PhysicsSystem::~PhysicsSystem()
 	BaseSystem::~BaseSystem();
 }
 
-void PhysicsSystem::update(Entity& t_entity/*float t_deltaTime*/) //deltaTime will be required here so it will need to be uncommented and used
+void PhysicsSystem::update(Entity& t_entity, float t_dt) //deltaTime will be required here so it will need to be uncommented and used
```
Comment (CathalRedmond) [inline ARGO/src/PhysicsSystem.cpp]: comment isn't needed any more

File: ARGO/src/ProjectileManager.cpp [hunk 9]
```diff
@@ -68,20 +78,36 @@ void ProjectileManager::createEnemyBullet(const createBulletEvent& t_event)
 	}
 }
 
-void ProjectileManager::update(BaseSystem* t_system)
+void ProjectileManager::update(float t_dt)
 {
 	for (auto& bullet : m_playerBullets)
 	{
 		if (static_cast<HealthComponent*>(bullet.entity.getAllComps().at(COMPONENT_ID::HEALTH_ID))->getHealth() > 0)
```
Comment (CathalRedmond) [inline ARGO/src/ProjectileManager.cpp]: This logic and the one for the For loop below can be made into a function where you pass in the specific bullet

File: ARGO/src/CollisionSystem.cpp (hunk 4 above)
Comment (CathalRedmond) [reply ARGO/src/CollisionSystem.cpp]: this

**Review by CathalRedmond — CHANGES_REQUESTED** [review]

**Pull request merged**